_posts/2024-09-27-learning-a-programming-language-with-generative-ai.md

```diff
@@ -0,0 +1,58 @@
+
+![The turing tes](/img/2024-09-27-learning-a-programming-language-with-generative-ai/turing-test.png)
+
+# How to learn with GenAI
+
+Myself, I used a learning by doing approach. There are however, many ways one can learn with GenAI. Another way to learn a new programming language could be to step by step letting ChatGPT talk about the language in question, including its syntax, variables, typing, data structures and so on. I, however, would not find this approach useful as there are already online-documentation and books for most programming languages that can be used in the traditional way, and that are also reliable. Regardless of the way, there are some practices that are good to take into consideration:
+
+- ChatGPT is probably the best option for learning. AI-tools for coding do not explain and tell theory and concepts as well as ChatGPT
+- If using ChatGPT, it is preferable to use the paid version and create a customized agent. That will be trained to focus as much as possible on the subject in question.
+- Fact check, fact check, fact check! GenAI is not all knowing and will make mistakes!
+- Do not over rely on GenAI. When the learning seems to be going well, there is always the risk that one will trust it too much. That might lead to forgetting of fact checking as well as passivity and decreased learning, as GenAI is doing most of the job.
+- If using ChatGPT, do not use the same chat window for too long. As ChatGPT is iterating over previous answers in the conversation, it will sooner or later start hallucinating.
+
+# Conclusion
+ 
+GenAI can be seen as a learning assistant or tutor. It is by no means an oracle or a guru. Just like a real tutor, it can make mistakes, and is even likely to do so. The information and answers it gives is gathered from the internet without any guarantees of fact checking or correctness. My tasks succeeded in the sense that I learned the basics of .NET and C#. In my opinion, using ChatGPT for learning a new programming language or framework might be beneficial when having the option of learning by doing. It may also be useful if one needs an explanation of a certain topic, as it is answering questions. One should, however, always consider the possibilities of flaws and hallucinations, and remember to not over rely on ChatGPT. It will certainly not replace traditional learning.
```

Comment:
"The information and answers it gives is gathered" -> "The information and answers it gives **are** gathered"

Reply:
corrected